#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *
from math import pi

PACKAGE = "camera_calibration_virtana"

gen = ParameterGenerator()

gen.add("x_trans", double_t,    0, "X position of the target in reference to the world frame", 0,  -10, 10)
gen.add("y_trans", double_t,    0, "Y position of the target in reference to the world frame", 0,  -10, 10)
gen.add("z_trans", double_t,    0, "Z position of the target in reference to the world frame", -0.5,  0, 10)
gen.add("roll", double_t,    0, "Roll of the target in reference to the world frame", 0,  -2*pi, 2*pi)
gen.add("pitch", double_t,    0, "Pitch of the target in reference to the world frame", 0,  -2*pi, 2*pi)
gen.add("yaw", double_t,    0, "Yaw of the target in reference to the world frame", 0,  -2*pi, 2*pi)
gen.add("link_name",    str_t,    0, "Name of link",  "calibration_target_link")

exit(gen.generate(PACKAGE, "world_transform_pub", "calibration"))
